src/query/storage/block.go

```diff
@@ -90,6 +90,38 @@ func (m *multiBlockWrapper) SeriesIter() (block.SeriesIter, error) {
 	return m.consolidated.SeriesIter()
 }
 
+func (m *multiBlockWrapper) UpdateMetas(
+	meta block.Metadata,
+	seriesMetas []block.SeriesMeta,
+) (block.Block, error) {
+	var (
+		consolidated   block.Block
+		unconsolidated block.UnconsolidatedBlock
+		err            error
+	)
+	if m.consolidated != nil {
```

Comment:
do you need to lock ?

Reply:
I suppose it can't hurt :P

Going to change it to a RWLock and just RLock this